### PR TITLE
Android main scheduler and sample app

### PR DESCRIPTION
**Pull request by arkivanov**

### CODE REVIEW

File: library/src/androidMain/kotlin/com/badoo/reaktive/scheduler/MainScheduler.kt [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    override fun newExecutor(): Scheduler.Executor =
+        ExecutorImpl()
+            .also(disposables::add)
+
+    override fun destroy() {
+        disposables.dispose()
+    }
+
+    private class ExecutorImpl : Scheduler.Executor {
+        @Volatile
```
Comment (olegpetsjonkin) [inline library/src/androidMain/kotlin/com/badoo/reaktive/scheduler/MainScheduler.kt]: Do we need Volatile here?

Comment (arkivanov) [reply library/src/androidMain/kotlin/com/badoo/reaktive/scheduler/MainScheduler.kt]: Absolutely, as it's accessed from isDisposed's get method where there is no synchronisation.

File: library/src/androidMain/kotlin/com/badoo/reaktive/scheduler/MainScheduler.kt [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+        override fun cancel() {
+            executeIfNotRecycled {
+                it.removeCallbacksAndMessages(null)
+            }
+        }
+
+        private inline fun <T> executeIfNotRecycled(block: (Handler) -> T): T? {
+            if (handler != null) {
+                synchronized(monitor) {
```
Comment (mk-jones) [inline library/src/androidMain/kotlin/com/badoo/reaktive/scheduler/MainScheduler.kt]: Why synch instead of a lock?

Comment (arkivanov) [reply library/src/androidMain/kotlin/com/badoo/reaktive/scheduler/MainScheduler.kt]: Because we have proper synchronisation in JVM

**Pull request merged**